app-libs/exchange-oracle/src/lib.rs

```diff
@@ -32,16 +32,13 @@ pub mod sgx_reexport_prelude {
 	pub use url_sgx as url;
 }
 
-use crate::error::Error;
+use crate::{error::Error, types::TradingPair};
 
 pub mod coingecko;
 pub mod error;
+pub mod types;
 
 pub trait GetExchangeRate {
 	/// Get the cryptocurrency/fiat_currency exchange rate
-	fn get_exchange_rate(
-		&mut self,
-		cryptocurrency: &str,
-		fiat_currency: &str,
-	) -> Result<f32, Error>;
+	fn get_exchange_rate(&mut self, trading_pair: TradingPair) -> Result<f32, Error>;
```

Comment:
I'd strongly recommend to define an exchange rate type

Like `type exchange_rate = f32`;

This makes it clear what this `f32` stands for and can be easily changed in case it's updated to a different type due to whatever reason.

Reply:
agreed, but not float! fixpoint U32F32